controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1131,6 +1137,363 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
+                               << std::endl
+                               << " not the same";
+                    return false;
+                }
+            }
+
+            LOG(DEBUG) << "recieved tlvReceiverLinkMetric from al_mac="
+                       << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+                       << "reported  al_mac =" << network_utils::mac_to_string(neighbor_al_mac)
+                       << std::endl;
+
+            //fill rx data from TLV
+            if (!new_link_metric_data.add_receiver_link_metric(RxLinkMetricData)) {
+                LOG(ERROR) << "adding RxLinkMetricData has failed";
+                return false;
+            }
+        } else {
+            LOG(ERROR) << " unexpected tlv is recieved ";
+            return false;
+        }
+    }
+
+    link_metric_data[reporting_agent_al_mac][reporting_agent_al_mac] = new_link_metric_data;
+
+    LOG(DEBUG) << " Added metric data from "
+               << " al_mac = " << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+               << "reported data for "
+               << " neighbor al_mac =" << network_utils::mac_to_string(neighbor_al_mac)
+               << std::endl;
+
+    print_link_metric_map(link_metric_data);
+
+    // auto agents = database.get_all_connected_ires();
+    auto agents = database.get_nodes(beerocks::TYPE_IRE);
+
+    for (auto &al_mac : agents) {
+        if (link_metric_data.find(net::network_utils::mac_from_string(al_mac)) ==
+            link_metric_data.end()) {
+            //If agent mac still not in database
+            return true;
+        }
+    }
+    //If all agents already in database, create Combined Metrics CMDU
+    // build Combined Metrics message
+    if (!cmdu_tx.create(0, ieee1905_1::eMessageType::COMBINED_INFRASTRUCTURE_METRICS_MESSAGE)) {
+        LOG(ERROR) << "cmdu creation of type COMBINED_INFRASTRUCTURE_METRICS_MESSAGE, has failed";
+        return false;
+    }
+
+    for (auto &agent : link_metric_data) {
+
+        for (auto &per_neighbor : link_metric_data[agent.first]) {
+
+            auto &vrx = per_neighbor.second.receiverLinkMetrics;
+            if (vrx.size()) {
+                auto link_metric_rx_tlv = cmdu_tx.addClass<ieee1905_1::tlvReceiverLinkMetric>();
+                link_metric_rx_tlv->reporter_al_mac() = agent.first;
+                link_metric_rx_tlv->neighbor_al_mac() = per_neighbor.first;
+                for (auto &interface_pair_info : vrx) {
+                    LOG(DEBUG) << "     interface_pair_info_length BEFORE ALLOC:  "
+                               << link_metric_rx_tlv->interface_pair_info_length();
+
+                    if (!link_metric_rx_tlv->alloc_interface_pair_info()) {
+                        LOG(ERROR) << "alloc_interface_pair_info() has failed!";
+                        return false;
+                    }
+
+                    auto interfeace_idx =
```

Comment:
This would be a lot simpler as:

```
if (!link_metric_rx_tlv->alloc_interface_pair_info(vrx.size())) {
    LOG(ERROR) << "alloc_interface_pair_info() has failed!";
    return false;
}
size_it interface_idx = 0;
for (auto &interface_pair_info : vrx) {
    auto interface_tuple =
        link_metric_rx_tlv->interface_pair_info(interface_idx);
    ...
    interface_idx++;
}
```

(btw, spelling mistake in interfeace_idx)